temporalio/worker/workflow_instance.py

```diff
@@ -1121,6 +1123,20 @@ async def _run_top_level_workflow_function(self, coro: Awaitable[None]) -> None:
                 f"Workflow raised failure with run ID {self._info.run_id}",
                 exc_info=True,
             )
+            # If a cancel was requested, and the failure is from an activity or
+            # child, and its cause was a cancellation, we want to use that cause
+            # instead because it means a cancel bubbled up while waiting on an
+            # activity or child.
+            if (
+                self._cancel_requested
+                and (
+                    isinstance(err, temporalio.exceptions.ActivityError)
+                    or isinstance(err, temporalio.exceptions.ChildWorkflowError)
+                )
+                and isinstance(err.cause, temporalio.exceptions.CancelledError)
+            ):
```

Comment:
Yeah, the auto-formatter we are forced to use makes this look ugly. It gets worse with a var:

```python
            activity_or_child = isinstance(
                err, temporalio.exceptions.ActivityError
            ) or isinstance(err, temporalio.exceptions.ChildWorkflowError)
            if (
                self._cancel_requested
                and activity_or_child
                and isinstance(err.cause, temporalio.exceptions.CancelledError)
            ):
                err = err.cause
```

The Python SDK looks _much_ uglier because of this formatter, but it's the industry norm. It's also why few-hundred line files look like thousands.

Reply:
+1 to not loving black but it is the standard now so 🤷